cmd/fleet/cron.go

```diff
@@ -153,142 +220,125 @@ func cronVulnerabilities(
-
-						level.Error(logger).Log("err", "triggering vulnerabilities webhook", "details", err)
-						sentry.CaptureException(err)
-					}
-
-				case "jira":
-					// queue job to create jira issues
-					if err := worker.QueueJiraJobs(
-						ctx,
-						ds,
-						kitlog.With(logger, "jira", "vulnerabilities"),
-						recentVulns,
-					); err != nil {
-						level.Error(logger).Log("err", "queueing vulnerabilities to jira", "details", err)
-						sentry.CaptureException(err)
-					}
-
-				case "zendesk":
-					// queue job to create zendesk ticket
-					if err := worker.QueueZendeskJobs(
-						ctx,
-						ds,
-						kitlog.With(logger, "zendesk", "vulnerabilities"),
-						recentVulns,
-					); err != nil {
-						level.Error(logger).Log("err", "queueing vulnerabilities to zendesk", "details", err)
-						sentry.CaptureException(err)
-					}
-
-				default:
-					err = errors.New("no vuln automations enabled")
-					level.Error(logger).Log("err", "attempting to process vuln automations", err)
+			case automation_zendesk:
+				if err := worker.QueueZendeskJobs(
+					ctx,
+					ds,
+					kitlog.With(logger, "zendesk", "vulnerabilities"),
+					recentVulns,
+				); err != nil {
+					level.Error(logger).Log("err", "queueing vulnerabilities to zendesk", "details", err)
 					sentry.CaptureException(err)
 				}
+			default:
+				err := fmt.Errorf("unknown automation: %d", automation)
+				level.Error(logger).Log("err", err)
+				sentry.CaptureException(err)
 			}
 		}
+	}
 
-		if err := ds.CalculateHostsPerSoftware(ctx, time.Now()); err != nil {
-			level.Error(logger).Log("msg", "calculating hosts count per software", "err", err)
+	if err := ds.CalculateHostsPerSoftware(ctx, time.Now()); err != nil {
+		level.Error(logger).Log("msg", "calculating hosts count per software", "err", err)
+		sentry.CaptureException(err)
+	}
+
+	// It's important vulnerabilities.PostProcess runs after ds.CalculateHostsPerSoftware
+	// because it cleans up any software that's not installed on the fleet (e.g. hosts removal,
+	// or software being uninstalled on hosts).
+	if !vulnDisabled {
+		if err := vulnerabilities.PostProcess(ctx, ds, vulnPath, logger, config); err != nil {
+			level.Error(logger).Log("msg", "post processing CVEs", "err", err)
 			sentry.CaptureException(err)
 		}
+	}
 
-		// It's important vulnerabilities.PostProcess runs after ds.CalculateHostsPerSoftware
-		// because it cleans up any software that's not installed on the fleet (e.g. hosts removal,
-		// or software being uninstalled on hosts).
-		if !vulnDisabled {
-			if err := vulnerabilities.PostProcess(ctx, ds, vulnPath, logger, config); err != nil {
-				level.Error(logger).Log("msg", "post processing CVEs", "err", err)
+	level.Debug(logger).Log("vulnerabilities", "done")
+	return nil
+}
+
+//go:generate go run golang.org/x/tools/cmd/stringer -type=automation
+type automation int
+
+const (
+	automation_none automation = iota
+	automation_webhook
+	automation_jira
+	automation_zendesk
```

Comment:
Nit: it would be more idiomatic to use `automationNone`, `automationWebhook`, etc. I implemented something similar while working on failing policies automation, I used string constants instead FWIW (removes the need to go:generate stringer).